py/server/tests/test_pyfunc_return_java_values.py

```diff
@@ -103,6 +105,62 @@ def test_datetime_scalar_return(self):
                 t = empty_table(10).update("X = i").update(f"Y= fn(X)")
                 self.assertEqual(t.columns[1].data_type, dtypes.Instant)
 
+    def test_datetime_array_return(self):
+
+        dt = datetime.datetime.now()
+        ts = pd.Timestamp(dt)
+        np_dt = np.datetime64(dt)
+        dt_list = [ts, np_dt, dt]
+
+        # test if we can convert to numpy datetime64 array
+        np_array = np.array([pd.Timestamp(dt).to_numpy() for dt in dt_list], dtype=np.datetime64)
+
+        dt_dtypes = [
+            "np.ndarray[np.dtype('datetime64[ns]')]",
+            "List[datetime.datetime]",
+            "Tuple[pd.Timestamp]"
+        ]
+
+        dt_data = [
+            "dt_list",
+            "np_array",
+        ]
+
+        for np_dtype in dt_dtypes:
+            for data in dt_data:
+                with self.subTest(np_dtype=np_dtype, data=data):
+                    func_decl_str = f"""def fn(col) -> {np_dtype}:"""
+                    # func_body_str_1 = f"""    nonlocal {data}"""
+                    func_body_str_1 = """"""
+                    func_body_str_2 = f"""    return {data}"""
```

Comment:
1) Is the commented line needed?
2) This test is funky.  I'm not sure what it is trying to do.  For example, the value actually returned is either a list of mixed types or a numpy array of datetimes. ... but the type hint matches the return value in only 1 of 6 cases.  If this is intentional, there needs to be a better comment about the intention of the test and what is going on.

Reply:
Additionally, this weird test points out that we can potentially convert a union type like:
```def fn(col) -> List[Union[datetime.datetime,pd.Timestamp,np.datetime64]]`
Does this work?  Should we try to support it?  Should we test it?

I'm guessing that the guts work but there are a few places that would need a change.

Reply:
I am not sure if we want to support the many combinations of type hints that certainly are valid but hard to cover completely. I don't think it is an unreasonable requirement for users to provide a more straightforward typehint for the return type, which is more Pythonic anyways. We currently parse it as just the generic object. I have added a test case .